### PR TITLE
use dfs instead of bfs in trie iterator

### PR DESCRIPTION
**Pull request by ssd04**
## Reasoning behind the pull request
- Use depth first search in order to try to optimize getting serialized nodes
  
## Proposed changes
- change trie search to use DFS

## Testing procedure
- Standard testing procedure

## Pre-requisites

Based on the [Contributing Guidelines](CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by BeniaminDrasovean — DISMISSED** [review]

File: trie/patriciaMerkleTrie.go [hunk 1]
```diff
@@ -398,7 +398,7 @@ func (tr *patriciaMerkleTrie) GetSerializedNodes(rootHash []byte, maxBuffToSend
 		return nil, 0, err
 	}
 
-	it, err := NewIterator(newTr)
+	it, err := NewBFSIterator(newTr)
```
Comment (sstanculeanu) [inline trie/patriciaMerkleTrie.go:401]:
```suggestion
	it, err := NewDFSIterator(newTr)
```

Comment (ssd04) [reply trie/patriciaMerkleTrie.go]: done

**Review by sstanculeanu — DISMISSED** [review]

**Review by BeniaminDrasovean — DISMISSED** [review]

File: trie/bfsIterator.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"github.com/ElrondNetwork/elrond-go/common"
+)
+
+type bfsIterator struct {
+	*baseIterator
+}
+
+// NewBFSIterator creates a new level order traversal iterator
+func NewBFSIterator(trie common.Trie) (*bfsIterator, error) {
```
Comment (iulianpascalau) [inline trie/bfsIterator.go:12]: Why do we've implemented this if will not go to use it anyway? 
Otherwise, I would go for a factory & config value

Comment (ssd04) [reply trie/bfsIterator.go]: removed the old bfs implementation

**Review by iulianpascalau — DISMISSED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed

**Pull request merged**